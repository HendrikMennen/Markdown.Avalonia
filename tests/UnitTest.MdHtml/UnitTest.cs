using ApprovalTests;
using ApprovalTests.Reporters;
using Avalonia.Controls;
using Markdown.Avalonia;
using Markdown.Avalonia.Html;
using NUnit.Framework;
using System.Linq;
using UnitTest.Base;
using UnitTest.Base.Utils;
using UnitTest.MdHtml.Test;

using MarkdownEngine = Markdown.Avalonia.Markdown;

namespace UnitTest.MdHtml
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest : UnitTestBase
    {
        public MarkdownEngine Engine
        {
            get
            {
                var plugins = new MdAvPlugins();
                plugins.Plugins.Add(new HtmlPlugin());

                return new MarkdownEngine()
                {
                    Plugins = plugins
                };
            }
        }

        [Test]
        [RunOnUI]
        public void Button()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void CodeBlock()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void InlineCode()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Input()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void List()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Progres()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Details()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void TypicalBlock()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void TypicalInline()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Mixing()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }
    }
}
