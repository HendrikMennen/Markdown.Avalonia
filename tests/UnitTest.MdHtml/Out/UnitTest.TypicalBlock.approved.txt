<?xml version="1.0" encoding="utf-16"?>
<StackPanel xmlns:c="clr-namespace:ColorTextBlock.Avalonia;assembly=ColorTextBlock.Avalonia" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns="https://github.com/avaloniaui">
  <c:CTextBlock Classes="Heading1">
    <c:CBold Classes="Bold">
      <c:CRun Text="Heading1 " />
    </c:CBold>
    <c:CRun Text=" Title" />
  </c:CTextBlock>
  <c:CTextBlock Classes="Heading2">
    <c:CItalic Classes="Italic">
      <c:CRun Text="Sub " />
    </c:CItalic>
    <c:CRun Text=" Titile" />
  </c:CTextBlock>
  <c:CTextBlock Classes="Heading3">
    <c:CRun Text="H3_Title" />
  </c:CTextBlock>
  <c:CTextBlock Classes="Heading4">
    <c:CRun Text="H4_Title" />
  </c:CTextBlock>
  <c:CTextBlock Classes="Heading5">
    <c:CRun Text="H5_Title" />
  </c:CTextBlock>
  <c:CTextBlock Classes="Heading6">
    <c:CRun Text="H6_Title" />
  </c:CTextBlock>
  <Border Classes="Address" xmlns="">
    <Border.Child>
      <StackPanel Classes="Address">
        <c:CTextBlock>
          <c:CRun Text="Written by " />
          <c:CHyperlink HoverBackground="{x:Null}" HoverForeground="{x:Null}" CommandParameter="mailto:webmaster@example.com" Classes="Hyperlink">
            <c:CRun Text="Jon Doe " />
          </c:CHyperlink>
          <c:CRun Text="." />
          <c:CLineBreak Text="&#xA;" />
          <c:CRun Text="simewhere, someland, somecountry" />
        </c:CTextBlock>
      </StackPanel>
    </Border.Child>
  </Border>
  <Border Classes="Article" xmlns="">
    <Border.Child>
      <StackPanel Classes="Article">
        <c:CTextBlock Classes="Heading3">
          <c:CRun Text="Article         Element" />
        </c:CTextBlock>
        <c:CTextBlock>
          <c:CRun Text="description of         this element..." />
        </c:CTextBlock>
      </StackPanel>
    </Border.Child>
  </Border>
  <Border Classes="Aside" xmlns="">
    <Border.Child>
      <StackPanel Classes="Aside">
        <c:CTextBlock Classes="Heading4">
          <c:CRun Text="ASide         Element" />
        </c:CTextBlock>
        <c:CTextBlock>
          <c:CRun Text="このタグに関する説明を書いてください・・・" />
        </c:CTextBlock>
      </StackPanel>
    </Border.Child>
  </Border>
  <Border Classes="Blockquote" xmlns="">
    <Border.Child>
      <StackPanel Classes="Blockquote">
        <Border Classes="Aside">
          <Border.Child>
            <StackPanel Classes="Aside">
              <c:CTextBlock Classes="Heading4">
                <c:CRun Text="ASide             Element" />
              </c:CTextBlock>
              <c:CTextBlock>
                <c:CRun Text="このタグに関する説明を書いてください・・・" />
              </c:CTextBlock>
            </StackPanel>
          </Border.Child>
        </Border>
        <c:CTextBlock>
          <c:CRun Text="hm?" />
        </c:CTextBlock>
      </StackPanel>
    </Border.Child>
  </Border>
  <Border Classes="Center" xmlns="">
    <Border.Child>
      <StackPanel Classes="Center">
        <c:CTextBlock Classes="Heading5">
          <c:CRun Text="Center Element" />
        </c:CTextBlock>
        <c:CTextBlock>
          <c:CRun Text="center tag center tag center tag..." />
        </c:CTextBlock>
      </StackPanel>
    </Border.Child>
  </Border>
  <c:CTextBlock Classes="Heading6">
    <c:CRun Text="H6_Title" />
  </c:CTextBlock>
  <c:CTextBlock>
    <c:CRun Text="abcdefghi" />
  </c:CTextBlock>
  <c:CTextBlock>
    <c:CRun Text="jklmnopqr" />
  </c:CTextBlock>
  <c:CTextBlock>
    <c:CRun Text="stuvwxyz" />
  </c:CTextBlock>
  <c:CTextBlock>
    <c:CRun Text="abcdefghi" />
  </c:CTextBlock>
  <c:CTextBlock>
    <c:CRun Text="jklmnopqr " />
    <c:CRun Text="stuvwxyz" />
  </c:CTextBlock>
</StackPanel>
